tests/native_tests/with_test.hy

```diff
@@ -34,12 +34,37 @@
         (assert (= t2 2))
         (assert (= t3 3))))
 
-  (defn test-quince-with []
-    "NATIVE: test four withs, one with no args"
-    (with [t1 (WithTest 1)
-           t2 (WithTest 2)
-           t3 (WithTest 3)
-           _ (WithTest 4)]
-          (assert (= t1 1))
-          (assert (= t2 2))
-          (assert (= t3 3))))
+(defn test-quince-with []
+  "NATIVE: test four withs, one with no args"
+  (with [t1 (WithTest 1)
+         t2 (WithTest 2)
+         t3 (WithTest 3)
+         _ (WithTest 4)]
+        (assert (= t1 1))
+        (assert (= t2 2))
+        (assert (= t3 3))))
+
+(defclass SuppressZDE [object]
+  (defn --enter-- [self])
+  (defn --exit-- [self exc-type exc-value traceback]
+    (and (not (none? exc-type)) (issubclass exc-type ZeroDivisionError))))
+
+(defn test-exception-suppressing-with []
+  ; https://github.com/hylang/hy/issues/1320
```

Comment:
> But, I don't think it's a good use of my time, or your time, to fuss over spacing and semicolons,
...
> This is why, although I too use Emacs, I don't use auto-indent for any language.

Turn on Emacs aggressive indent mode and you won't have to waste my time fussing over it.

> and I don't think any hard-and-fast rules about code formatting should be enforced. Rules that are useful most of the time are occasionally going to be less than useful
...
> But I don't want to spend time editing every PR for style issues that are more about tradition and fussy editors than about human legibility.

Lisp must be indented correctly to be legible. That particular rule is absolute. The machine doesn't care, but if humans have to maintain it, indent it properly. Code in a team project is read more than it is written. Don't create more work for everyone that follows.

We really should have an automated Travis check for proper indentation of Hy. But since Hy is young, we haven't gotten around to writing it.

> And some rules have no discernible value at all, like this one about one versus two semicolons.

You don't get to decide this unilaterally. I don't want to break the semicolon convention here without a good reason, and enough approval from Hylang's core to change the style guide to say that we're breaking with Lisp convention on this point. If you get that, then we can do one semicolon everywhere after the change. Not before.

It does have a use in Emacs. It will automatically indent it differently in a way that is useful. One semicolon is for [margin comments](https://stackoverflow.com/a/8825101). Over multiple lines it would take more work to align them if emacs didn't do it for you. Two semicolons automatically aligns with the other list elements. Three or more are never indented.